### PR TITLE
chore: reusable base page

### LINKED ISSUE #7739: Troubleshooting should be a DetailsPage
### Is your enhancement related to a problem? Please describe

Now that the FormPage has a single background color and no horizontal line, the Troubleshooting page looks odd because the tabs are floating.

Screenshot 2024-06-19 at 8 41 41 AM

### Describe the solution you'd like

Looks like this page should be a DetailsPage, which would also allow removing some of the 'duplicate' titles.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Provides a single base layout page that has enough options to support the current FormPage and DetailsPages.

This is enough that both Help and Troubleshooting pages can use this new layout instead of FormPage. This means they need to handle the close directly (maybe it's worth another page in the middle to handle this for both?), but it means we can remove the showBreadcrumb option from FormPage to start to clean it up.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

First part of #7757. (NavPage, SettingsPage, or merging in any other pages is not attempted here)

Fixes #7739.

### How to test this PR?

Check several form pages, details pages, troubleshooting, and help for regressions/change, should be none.

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/ui/src/lib/layouts/BasePage.svelte [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+export let subtitle: string | undefined = undefined;
+export let breadcrumbLeftPart: string | undefined = undefined;
+export let breadcrumbRightPart: string | undefined = undefined;
+export let breadcrumbTitle: string | undefined = '';
+export let hasClose: boolean = true;
+export let inProgress: boolean = false;
+
+let showBreadcrumb = breadcrumbLeftPart ?? breadcrumbRightPart;
+
+const dispatchClose = createEventDispatcher<{ close: undefined }>();
```
Comment (axel7083) [inline packages/ui/src/lib/layouts/BasePage.svelte]: We are not suppose to use dispatcher with svelte 5

File: packages/ui/src/lib/layouts/BasePage.svelte [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import { LinearProgress } from '..';
+import CloseButton from '../button/CloseButton.svelte';
+import Link from '../link/Link.svelte';
+
+export let title: string;
+export let titleDetail: string | undefined = undefined;
+export let subtitle: string | undefined = undefined;
+export let breadcrumbLeftPart: string | undefined = undefined;
+export let breadcrumbRightPart: string | undefined = undefined;
+export let breadcrumbTitle: string | undefined = '';
```
Comment (axel7083) [inline packages/ui/src/lib/layouts/BasePage.svelte]: Not related to this PR but I know that `left | right` breadcrumb has been used in other component, but I really don't like it, as it limit to only two when we could need more ?

**Comment by deboer-tim** [comment]
Rebased and made a couple fixes/simplifications.

File: packages/ui/src/lib/layouts/BasePage.svelte (hunk 1 above)
Comment (deboer-tim) [reply packages/ui/src/lib/layouts/BasePage.svelte]: This PR is merging the dispatcher from two components into a single parent, IMHO that would make reviewing harder and should be a separate PR.

File: packages/ui/src/lib/layouts/BasePage.svelte (hunk 2 above)
Comment (deboer-tim) [reply packages/ui/src/lib/layouts/BasePage.svelte]: I agree, but this PR is just merging two copies of that breadcrumb code into one parent, not trying to improve it at this point.

**Review by axel7083 — APPROVED** [review]

**Comment by benoitf** [comment]
I would rename `BasePage` to just `Page`

**Comment by deboer-tim** [comment]
> I would rename `BasePage` to just `Page`

Yeah, that's better, renamed.

**Review by benoitf — APPROVED** [review]

**Review by cdrage — APPROVED** [review]

**Pull request merged**